src/views/Stake/components/ExternalStakePools/hooks/useStakePoolTVL.ts

```diff
@@ -0,0 +1,52 @@
+import { useQuery } from "react-query";
+import { GOHM_ADDRESSES } from "src/constants/addresses";
+import { createDependentQuery, getTokenPrice, parseBigNumber, queryAssertion } from "src/helpers";
+import { useStaticPairContract } from "src/hooks/useContract";
+import { useGohmPrice } from "src/hooks/usePrices";
+import { useStaticProvider } from "src/hooks/useStaticProvider";
+import { ExternalPool } from "src/lib/ExternalPool";
+
+export const stakePoolTVLQueryKey = (poolAddress: string) => ["useStakePoolTVL", poolAddress];
+
+export const useStakePoolTVL = (pool: ExternalPool) => {
+  const provider = useStaticProvider(pool.networkID);
+  const contract = useStaticPairContract(pool.address, provider);
+
+  const useDependentQuery = createDependentQuery(stakePoolTVLQueryKey(pool.address));
+
+  // Get dependent data in parallel
+  const { data: gohmPrice } = useGohmPrice();
+  const reserves = useDependentQuery("reserves", () => contract.getReserves());
+  const firstTokenAddress = useDependentQuery("firstTokenAddress", () => contract.token0());
+  const poolTokenSupply = useDependentQuery("poolTokenSupply", () => contract.totalSupply());
+  const stakedBalance = useDependentQuery("stakedBalance", () => contract.balanceOf(pool.masterchef));
+  const nonGohmTokenPrice = useDependentQuery("nonGohmTokenPrice", () => getTokenPrice(pool.pairGecko));
```

Comment:
wow, I hadn't seen `dependent queries` before. nice

Reply:
Cleanest way I could come up with for downloading all the other data that a query needs. And ensures requests are done in parallel, as soon as we know we need that data.